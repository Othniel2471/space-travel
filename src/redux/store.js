import { configureStore } from '@reduxjs/toolkit';
import RocketSlice from './rocket/RocketSlice';

const store = configureStore({
  reducer: {
    rockets: RocketSlice,
  },
});

export default store;
