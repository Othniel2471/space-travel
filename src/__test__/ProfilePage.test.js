import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Profile from '../pages/ProfilePage';

describe('Component testing', () => {
  test('Profile component should render correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Profile />
        </Router>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
