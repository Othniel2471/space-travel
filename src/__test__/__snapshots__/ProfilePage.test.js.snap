// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing Profile component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="container mx-auto px-3 py-3 flex gap-10 flex-col md:flex-row"
  >
    <div
      class="w-6/12"
    >
      <h3
        class="text-lg font-semibold"
      >
        My Missions
      </h3>
      <p
        class="border border-gray-300 px-4 py-3 text-left"
      >
        No Missions
      </p>
    </div>
    <div
      class="w-6/12"
    >
      <h3
        class="text-lg font-semibold"
      >
        My Rockets
      </h3>
      <p
        class="border border-gray-300 px-4 py-3 text-left"
      >
        No Rockets
      </p>
    </div>
  </div>
</DocumentFragment>
`;
